### PR TITLE
Configure gsoci.azurecr.io as the registry to use by default

### PR DESCRIPTION
**Pull request by marians**
Towards

- https://github.com/giantswarm/roadmap/issues/3017

This PR replaces any occurrence of `docker.io` in `values.yaml` files with `gsoci.azurecr.io`,
to make the new ACR registry the default one.

### CODE REVIEW

**Comment by marians** [comment]
I can't explain why the schema check fails here.

File: helm/deletion-blocker-operator/values.schema.json [hunk 1]
```diff
@@ -156,7 +156,15 @@
...
+                    "type": "object",
+                    "properties": {
+                        "drop": {
+                            "type": "array",
+                            "items": {
+                                "type": "string"
+                            }
+                        }
+                    }
                 },
```
Comment (marians) [inline helm/deletion-blocker-operator/values.schema.json:168]: This change is not related to the purpose of the PR. I ran `helm-schema-gen` to find anything I could update about the schema, to please our stupid schema check, and found this missing chunk.

**Review by erkanerol — APPROVED** [review]
